memgpt/cli/cli_load.py

```diff
@@ -135,39 +105,56 @@ def store_docs(name, docs, user_id=None, show_progress=True):
     # compute and record passages
     embed_model = embedding_model(config.default_embedding_config)
 
+    storage = StorageConnector.get_storage_connector(TableType.DOCUMENTS, config, user_id)
+    docs_storage = []
+    for doc in docs:
+        doc_storage = Document(user_id=user_id, text=doc.text, id=uuid.UUID(doc.doc_id), data_source=data_source.name)
```

Comment:
@Maximilian-Winter I may be misunderstanding llama index, but wouldn't this still store the chunked text of a single file, rather than the text of the entire file?

Reply:
No, I don't think so, we are passing the results of the different llama index readers to this function, as far as I could see all readers return a list of documents. These documents represent the complete documents found by the llama index readers, in form of instances of the llama index Document class. The chunks get only created when they are added to the vector index. Where they get  chunked into nodes and get embedded.

Here is the code I use to access and get the chunks and their text and embeddings.
```python
 index = VectorStoreIndex.from_documents(docs, service_context=service_context, show_progress=True)
    doc_info = index.ref_doc_info
    passages = []
    doc_dict = {}
    passages_dict = index.docstore.docs

    # SimpleVectorStore has a get method and takes in the node id to retrieve the embedding.
    if isinstance(index.vector_store, SimpleVectorStore):
        simple_store = index.vector_store
        for doc_id, data in doc_info.items():
            nodes_dict = []
            for node_id in data.node_ids:
                nodes_dict.append((node_id, simple_store.get(node_id), passages_dict[node_id].to_dict()))
            doc_dict[doc_id] = nodes_dict
```

Reply:
This here access the chunked documents: passages_dict = index.docstore.docs
Which I access by using the node id found in doc_info. I get doc_info by accessing ref_doc_info which is a dictionary of doc_ids to node ids.

Reply:
I also checked the created sqlite database and it contains the unchunked complete text.